### PR TITLE
Create new app

### PR DESCRIPTION
**Pull request by qxz2mqe**
https://jira.catena-x.net/browse/CPLP-827

### CODE REVIEW

File: src/marketplace/CatenaX.NetworkServices.App.Service/InputModels/AppRequestModel.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+public class AppRequestModel
+{
+    /// <summary>
+    /// Private constructor.
+    /// </summary>
+    private AppRequestModel()
+    {
+        Provider = string.Empty;
+        Price = string.Empty;
+        UseCaseIds = new HashSet<Guid>();
```
Comment (ntruchsess) [inline src/marketplace/CatenaX.NetworkServices.App.Service/InputModels/AppRequestModel.cs:17]: What is this good for? A non-nullable property of a data-transfer-object shall be initialized as mandatory argument to the public constructor

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs [hunk 2]
```diff
@@ -240,4 +240,44 @@ public async Task<Guid> CreateAppAsync(AppInputModel appInputModel)
 
         return appId;
     }
+    
+    /// <inheritdoc/>
+    public async Task<Guid> AddAppAsync(AppRequestModel appRequestModel)
+    {
+        if (appRequestModel.Descriptions.Select(x => x.LanguageCode) == null)
```
[the remark this answers is not in the document]
Comment (qxz2mqe) [reply src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs]: fixed

File: src/marketplace/CatenaX.NetworkServices.App.Service/InputModels/AppRequestModel.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace CatenaX.NetworkServices.App.Service.InputModels;
+
+/// <summary>
+/// Model for requesting creation of an application.
+/// </summary>
+public class AppRequestModel
```
[the remark this answers is not in the document]
Comment (qxz2mqe) [reply src/marketplace/CatenaX.NetworkServices.App.Service/InputModels/AppRequestModel.cs]: fixed

File: src/marketplace/CatenaX.NetworkServices.App.Service/InputModels/AppRequestModel.cs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    /// <summary>
+    /// ID of the app's providing company.
+    /// </summary>
+    public Guid? ProviderCompanyId { get; set; }
+
+    /// <summary>
+    /// IDs of app's use cases.
+    /// </summary>
+    public virtual ICollection<Guid> UseCaseIds { get; set; }
```
[the remark this answers is not in the document]
Comment (qxz2mqe) [reply src/marketplace/CatenaX.NetworkServices.App.Service/InputModels/AppRequestModel.cs]: fixed-not require as it is moved to record

File: src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs [hunk 5]
```diff
@@ -240,4 +240,44 @@ public async Task<Guid> CreateAppAsync(AppInputModel appInputModel)
...
+        {
+            throw new NotFoundException($"Language Code does not exist");
```
[the remark this answers is not in the document]
Comment (qxz2mqe) [reply src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppsBusinessLogic.cs]: fixed

File: src/marketplace/CatenaX.NetworkServices.App.Service/InputModels/AppRequestModel.cs (hunk 1 above)
Comment (qxz2mqe) [reply src/marketplace/CatenaX.NetworkServices.App.Service/InputModels/AppRequestModel.cs]: fixed:- not require as it is moved to record

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**